stan/math/rev/mat/fun/sqrt_spd.hpp

```diff
@@ -35,19 +35,23 @@ class sqrt_spd_op {
   //  derivative-or-differential-of-symmetric-square-root-of-a-matrix
 
   template <std::size_t size>
-  std::tuple<Eigen::VectorXd>
+  std::tuple<Eigen::MatrixXd>
   multiply_adjoint_jacobian(const std::array<bool, size> &needs_adj,
-                            const Eigen::VectorXd &adj) const {
+                            const Eigen::MatrixXd &adj) const {
```

Comment:
Thanks for fixing this, but I need to unconfuse myself before I tackle the rev specialization of `inv_sqrt_spd`. So, I see `adj` is a `MatrixXd` now but what are its dimensions and what is in it? I thought it made sense when it was a `VectorXd` of size `K_^2` of sensitivities of the scalar objective with respect to each of the `K_^2` elements of `sqrt_spd(m)` and by multiplying it by the Jacobian we get sensitivities with respect to the `K_^2` elements of `m`. But my thinking could not have been right if it was asserting before and now works (even for `K = 9` in the tests).

Reply:
If the output of the function is the matrix y_ij and we're autodiffing the variable L, then adj is the matrix dL/dy_ij.

It has the same shape as the output of operator(), but, instead of being values of operator(), it's been packed with the adjoints with respect to those values.

Reply:
But if `adj` is a KxK matrix, how is this working when the inverse Jacobian is K^2 x K^2? The inverse Jacobian is the `(kroneckerProduct(sqrt_m, MatrixXd::Identity(K_, K_)) + kroneckerProduct(MatrixXd::Identity(K_, K_), sqrt_m))` which is the Kronecker sum thing derived on MathOverflow. It would make more sense to me if `adj` was getting vectorized to be `K^2 x 1`.

Reply:
> adj was getting vectorized to be K^2 x 1.

This is what the Map<VectorXd> is doing. It's taking the KxK adj and then looking at it likes it's a K^2x1 vector.